src/midi/midi_fluidsynth.cpp

```diff
@@ -246,27 +246,48 @@ bool MidiHandlerFluidsynth::Open(MAYBE_UNUSED const char *conf)
 		LOG_MSG("MIDI: Using SoundFont '%s' with levels attenuated by %d%%",
 		        soundfont.c_str(), scale_by_percent);
 
+	constexpr int fx_group = -1; // applies setting to all groups
+
 	// Use a 7th-order (highest) polynomial to generate MIDI channel waveforms
-	constexpr int all_channels = -1;
-	fluid_synth_set_interp_method(fluid_synth.get(), all_channels,
+	fluid_synth_set_interp_method(fluid_synth.get(), fx_group,
 	                              FLUID_INTERP_HIGHEST);
 
-	// Apply reasonable chorus and reverb settings matching ScummVM's defaults
+	// Use reasonable chorus and reverb settings matching ScummVM's defaults
 	constexpr int chorus_number = 3;
 	constexpr double chorus_level = 1.2;
 	constexpr double chorus_speed = 0.3;
 	constexpr double chorus_depth = 8.0;
-	fluid_synth_set_chorus_on(fluid_synth.get(), 1);
-	fluid_synth_set_chorus(fluid_synth.get(), chorus_number, chorus_level,
-	                       chorus_speed, chorus_depth, FLUID_CHORUS_MOD_SINE);
-
+	constexpr auto sine = fluid_chorus_mod::FLUID_CHORUS_MOD_SINE;
 	constexpr double reverb_room_size = 0.61;
 	constexpr double reverb_damping = 0.23;
 	constexpr double reverb_width = 0.76;
 	constexpr double reverb_level = 0.56;
+
+// current API calls as of 2.2
+#if FLUIDSYNTH_VERSION_MINOR >= 2
```

Comment:
I don't like that we use two different APIs depending on the library we have on build machine…

I suggest upgrading the minimum required version of FluidSynth to 2.1 and using new API only.

I looked through the repology page and it seems like all serious distributions that package FluiSynth 2.x, have at least version 2.1.0 :)

Reply:
> I suggest upgrading the minimum required version of FluidSynth to 2.1 and using new API only.

I started with this approach; here's the first push to this branch today:

https://github.com/dosbox-staging/dosbox-staging/commit/71f8efc1deb6059a48c43ffa6d93ac050dd661de

It compiled perfectly fine on my home system with FluidSynth 2.1.3 from repo, as well as using the latest 2.2.0 from the static builder -- both without warnings.

However, on macOS in the CI environment, which is running FluidSynth 2.1.7, it actually bombs out saying it's not even aware of these calls:

https://github.com/dosbox-staging/dosbox-staging/runs/1855941103?check_suite_focus=true

This was rather confusing, but it seems on macOS the "old" calls are still valid for 2.1.x.  This is using the brew-installed version.

That said - this same FluidSynth library was the one that broke for users at runtime back when we used it for release builds. 

I'll retry this effort, but this time put the static builder into actuion and use it's headers and libraries.  If this works, then it bodes well for meson because we're planning the same setup there :+1: